src/main/scala/com/databricks/spark/avro/DefaultSource.scala

```diff
@@ -16,16 +16,210 @@
 
 package com.databricks.spark.avro
 
-import org.apache.spark.sql.SQLContext
-import org.apache.spark.sql.sources.{HadoopFsRelation, HadoopFsRelationProvider}
+import java.io.{FileNotFoundException, IOException, ObjectInputStream, ObjectOutputStream}
+import java.net.URI
+import java.util.zip.Deflater
+
+import scala.util.control.NonFatal
+
+import com.databricks.spark.avro.DefaultSource.{IgnoreFilesWithoutExtensionProperty, SerializableConfiguration}
+import org.apache.avro.SchemaBuilder
+import org.apache.avro.file.{DataFileConstants, DataFileReader}
+import org.apache.avro.generic.{GenericDatumReader, GenericRecord}
+import org.apache.avro.mapred.{AvroOutputFormat, FsInput}
+import org.apache.avro.mapreduce.AvroJob
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileStatus, Path}
+import org.apache.hadoop.mapreduce.Job
+import org.slf4j.LoggerFactory
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.encoders.RowEncoder
+import org.apache.spark.sql.catalyst.expressions.GenericRow
+import org.apache.spark.sql.execution.datasources.{FileFormat, OutputWriterFactory, PartitionedFile}
+import org.apache.spark.sql.sources.{DataSourceRegister, Filter}
 import org.apache.spark.sql.types.StructType
 
-private[avro] class DefaultSource extends HadoopFsRelationProvider {
+private[avro] class DefaultSource extends FileFormat with DataSourceRegister {
```

Comment:
Actually we can't... `DefaultSource` is the class name that we are using to lookup concrete data source implementations. The reason why we can rename those built-in data sources is that they are [special cased][1] (which can be a little bit misleading).

[1]: https://github.com/apache/spark/blob/7b9071eeaa62fd9a51d9e94cfd479224b8341517/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala#L82-L109